.github/scripts/branch_pr_issue_closer.py

```diff
@@ -135,176 +98,165 @@ def _main_prog():
 
-    #Extract list of open issues from repo:
-    open_repo_issues = cam_repo.get_issues(state='open')
+        #Extract list of open issues from repo:
+        open_repo_issues = cam_repo.get_issues(state='open')
 
-    #Collect all open repo issues:
-    open_issues = [issue.number for issue in open_repo_issues]
+        #Collect all open repo issues:
+        open_issues = [issue.number for issue in open_repo_issues]
 
-    #+++++++++++++++++++++++++++++++++++++++++++++
-    #Create integer list of all open pull requests
-    #+++++++++++++++++++++++++++++++++++++++++++++
+        #+++++++++++++++++++++++++++++++++++++++++++++
+        #Create integer list of all open pull requests
+        #+++++++++++++++++++++++++++++++++++++++++++++
 
-    #Extract list of open PRs from repo:
-    open_repo_pulls = cam_repo.get_pulls(state='open')
+        #Extract list of open PRs from repo:
+        open_repo_pulls = cam_repo.get_pulls(state='open')
 
-    #Collect all open pull requests:
-    open_pulls = [pr.number for pr in open_repo_pulls]
+        #Collect all open pull requests:
+        open_pulls = [pr.number for pr in open_repo_pulls]
 
-    #+++++++++++++++++++++++++++++++++++++++++++++++++
-    #Check if one of the keywords exists in PR message
-    #+++++++++++++++++++++++++++++++++++++++++++++++++
+        #+++++++++++++++++++++++++++++++++++++++++++++++++
+        #Check if one of the keywords exists in PR message
+        #+++++++++++++++++++++++++++++++++++++++++++++++++
 
-    #Keywords are:
-    #close, closes, closed
-    #fix, fixes, fixed
-    #resolve, resolves, resolved
+        #Keywords are:
+        #close, closes, closed
+        #fix, fixes, fixed
+        #resolve, resolves, resolved
 
-    #Create regex pattern to find keywords:
-    keyword_pattern = re.compile(r'(^|\s)close(\s|s\s|d\s)|(^|\s)fix(\s|es\s|ed\s)|(^|\s)resolve(\s|s\s|d\s)')
+        #Create regex pattern to find keywords:
+        keyword_pattern = re.compile(r'(^|\s)close(\s|s\s|d\s)|(^|\s)fix(\s|es\s|ed\s)|(^|\s)resolve(\s|s\s|d\s)')
 
-    #Extract (lower case) Pull Request message:
-    pr_msg_lower = merged_pull.body.lower()
+        #Extract (lower case) Pull Request message:
+        pr_msg_lower = merged_pull.body.lower()
 
-    #search for at least one keyword:
-    if keyword_pattern.search(pr_msg_lower) is not None:
-        #If at least one keyword is found, then determine location of every keyword instance:
-        word_matches = keyword_pattern.finditer(pr_msg_lower)
-    else:
-        endmsg = "Pull request was merged without using any of the keywords.  Thus there are no issues to close."
-        end_script(endmsg)
+        #search for at least one keyword:
+        word_matches = []
+        if keyword_pattern.search(pr_msg_lower) is not None:
+            #If at least one keyword is found, then determine location of every keyword instance:
+            word_matches = keyword_pattern.finditer(pr_msg_lower)
+        else:
```

Comment:
```suggestion
        word_matches = keyword_pattern.finditer(pr_msg_lower)
        if word_matches is None:
```
Is there any benefit to only doing a `search` first if we have to `finditer` anyway if there's a match?  I imagine this will scan the entire message string anyway as GitHub defaults to putting the keywords at the bottom of the PR message.